pyccel/parser/semantic.py

```diff
@@ -1037,20 +1040,25 @@ def get_type_description(self, var, include_rank = True):
             descr += f'[{dims}]'
         return descr
 
-    def _check_argument_compatibility(self, input_args, func_args, expr, elemental):
+    def _check_argument_compatibility(self, input_args, func_args, expr, elemental, error=True):
         """
-        Check that the provided arguments match the expected types
+        Check that the provided arguments match the expected types.
 
         Parameters
         ----------
         input_args : list
-                     The arguments provided to the function
+                     The arguments provided to the function.
         func_args  : list
-                     The arguments expected by the function
+                     The arguments expected by the function.
         expr       : TypedAstNode
-                     The expression where this call is found (used for error output)
+                     The expression where this call is found (used for error output).
         elemental  : bool
-                     Indicates if the function is elemental
+                     Indicates if the function is elemental.
```

Comment:
NumpyDoc detects argument names by splitting at ` : ` so it will see an argument called "expr      " and raise an error.
```
        input_args : list
            The arguments provided to the function.
        func_args : list
            The arguments expected by the function.
        expr : TypedAstNode
            The expression where this call is found (used for error output).
        elemental : bool
            Indicates if the function is elemental.
```